Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift

```diff
@@ -574,6 +574,31 @@ extension SwiftPMBuildSystem: SKCore.BuildSystem {
       "--disable-index-store",
       "--target", target.targetID,
     ]
+    if self.toolsBuildParameters.configuration != self.destinationBuildParameters.configuration {
+      logger.fault(
+        """
+        Preparation is assuming that tools and destination are built using the same configuration, \
+        got tools: \(String(describing: self.toolsBuildParameters.configuration), privacy: .public), \
+        destination: \(String(describing: self.destinationBuildParameters.configuration), privacy: .public)
+        """
+      )
+    }
+    arguments += ["-c", self.destinationBuildParameters.configuration.rawValue]
+    if self.toolsBuildParameters.flags != self.destinationBuildParameters.flags {
+      logger.fault(
+        """
+        Preparation is assuming that tools and destination are built using the same build flags, \
+        got tools: \(String(describing: self.toolsBuildParameters.flags)), \
+        destination: \(String(describing: self.destinationBuildParameters.configuration))
+        """
+      )
+      logger.fault("Preparation is assuming that tools and destination to have the same build flags")
+    }
+    arguments += self.destinationBuildParameters.flags.cCompilerFlags.flatMap { ["-Xcc", $0] }
+    arguments += self.destinationBuildParameters.flags.cxxCompilerFlags.flatMap { ["-Xcxx", $0] }
+    arguments += self.destinationBuildParameters.flags.swiftCompilerFlags.flatMap { ["-Xswiftc", $0] }
+    arguments += self.destinationBuildParameters.flags.linkerFlags.flatMap { ["-Xlinker", $0] }
+    arguments += self.destinationBuildParameters.flags.xcbuildFlags?.flatMap { ["-Xxcbuild", $0] } ?? []
```

Comment:
Why are these needed? The prepare should already have these shouldn't it?

Reply:
These are flags that you pass when launching SourceKit-LSP and that are equivalent to the `-Xswiftc` flags passed to `swift build`. For example, if you need to build using `swift build -Xcxx -I/usr/lib/swift -Xcxx -I/usr/lib/swift/Block` then you should launch sourcekit-lsp using `sourcekit-lsp -Xcxx -I/usr/lib/swift -Xcxx -I/usr/lib/swift/Block`, which will then now also pass them to the `swift build` commands it runs as part of background indexing.